### ISSUE TITLE
Custom Cross-Validation Parameters in ForecastingGridSearchCV Cannot Be Tuned via param_grid

### ISSUE THREAD

**Issue by ebbalseven** [sktime/sktime#6062]
Problem Description:
When attempting to use ForecastingGridSearchCV for hyperparameter tuning in forecasting tasks, it appears that it's not possible to tune cross-validation parameters (such as window_length or step_length of SlidingWindowSplitter) directly via the param_grid.

code: 
from sktime.forecasting.arima import ARIMA
from sktime.forecasting.compose import TransformedTargetForecaster
from sktime.transformations.series.detrend import Deseasonalizer
from sktime.forecasting.base import ForecastingHorizon
from sktime.datasets import load_airline
from sktime.forecasting.model_selection import ForecastingGridSearchCV, SlidingWindowSplitter
from sktime.forecasting.model_selection import temporal_train_test_split

y = load_airline()
y_train, y_test = temporal_train_test_split(y, test_size=36)
fh = ForecastingHorizon(y_test.index, is_relative=False)

forecaster = TransformedTargetForecaster(
    [
        ("deseasonalize", Deseasonalizer()),
        ("ARIMA", ARIMA()),
    ]
)

param_grid = {"cv__step_length": [1, 2]}
cv = SlidingWindowSplitter(initial_window=int(len(y_train) * 0.8), window_length=20)
gscv = ForecastingGridSearchCV(forecaster, strategy="refit", cv=cv, param_grid=param_grid)
gscv.fit(y_train)

Expected Behavior:
The ability to tune cross-validation splitter parameters (e.g., step_length for SlidingWindowSplitter) directly via param_grid in ForecastingGridSearchCV.

Encountered Issue:
Received a ValueError indicating an invalid parameter for cv

ValueError: Invalid parameter cv for object TransformedTargetForecaster(steps=[('deseasonalize', Deseasonalizer()), ('ARIMA', ARIMA())]). Check the list of available parameters with `object.get_params().keys()`.

Is there any planned support for this feature?

thanks

**Comment by ebbalseven** [comment]
I am utilizing sktime for a time series forecasting task and wish to optimize my forecasting model by tuning cross-validation (CV) splitter parameters, specifically the window_length parameter of SlidingWindowSplitter, through ForecastingGridSearchCV. My goal is to dynamically adjust these parameters during the grid search process to identify the best model configuration. However, I encountered a limitation where param_grid does not support tuning CV splitter parameters directly  

example code : 
from sktime.forecasting.arima import ARIMA
from sktime.forecasting.compose import TransformedTargetForecaster
from sktime.transformations.series.detrend import Deseasonalizer
from sktime.forecasting.model_selection import ForecastingGridSearchCV, SlidingWindowSplitter
from sktime.datasets import load_airline
from sktime.forecasting.model_selection import temporal_train_test_split
import numpy as np

# Load dataset and prepare training and test sets
y = load_airline()
y_train, y_test = temporal_train_test_split(y, test_size=36)
fh = np.arange(1, 6)

# Define forecaster
forecaster = TransformedTargetForecaster([
    ("deseasonalize", Deseasonalizer()),
    ("ARIMA", ARIMA()),
])

# Attempt to define param_grid including CV splitter parameters
param_grid = {"window_length": [7, 12, 15]}
cv = SlidingWindowSplitter(initial_window=int(len(y_train) * 0.8), window_length=20)

# Initialize and fit ForecastingGridSearchCV
gscv = ForecastingGridSearchCV(forecaster, strategy="refit", cv=cv, param_grid=param_grid)
gscv.fit(y_train)


output: 
ValueError: Invalid parameter window_length for object TransformedTargetForecaster(steps=[('deseasonalize', Deseasonalizer()),
                                   ('ARIMA', ARIMA())]). Check the list of available parameters with `object.get_params().keys()`.

**Comment by ebbalseven** [comment]
I am utilizing sktime for a time series forecasting task and wish to optimize my forecasting model by tuning cross-validation (CV) splitter parameters, specifically the window_length parameter of SlidingWindowSplitter, through ForecastingGridSearchCV. My goal is to dynamically adjust these parameters during the grid search process to identify the best model configuration. However, I encountered a limitation where param_grid does not support tuning CV splitter parameters directly

example code :
from sktime.forecasting.arima import ARIMA
from sktime.forecasting.compose import TransformedTargetForecaster
from sktime.transformations.series.detrend import Deseasonalizer
from sktime.forecasting.model_selection import ForecastingGridSearchCV, SlidingWindowSplitter
from sktime.datasets import load_airline
from sktime.forecasting.model_selection import temporal_train_test_split
import numpy as np


y = load_airline()
y_train, y_test = temporal_train_test_split(y, test_size=36)
fh = np.arange(1, 6)


forecaster = TransformedTargetForecaster([
("deseasonalize", Deseasonalizer()),
("ARIMA", ARIMA()),
])


param_grid = {"window_length": [7, 12, 15]}
cv = SlidingWindowSplitter(initial_window=int(len(y_train) * 0.8), window_length=20)


gscv = ForecastingGridSearchCV(forecaster, strategy="refit", cv=cv, param_grid=param_grid)
gscv.fit(y_train)

output:
ValueError: Invalid parameter window_length for object TransformedTargetForecaster(steps=[('deseasonalize', Deseasonalizer()),
('ARIMA', ARIMA())]). Check the list of available parameters with object.get_params().keys().

**Comment by fkiraly** [comment]
could you describe the algorithm you would like to carry out?

Specifically, how would you compute the score for different `cv` settings?

**Comment by ebbalseven** [comment]
I just want to tune SlidingWindowSplitter window_length parameter in GridSearch

**Comment by fkiraly** [comment]
> I just want to tune SlidingWindowSplitter window_length parameter in GridSearch

Yes, I get that, but what I do not get is, how, algorithmically,

I would very much appreciate your explanation here. I.e., what's the abstract algorithm that you would like to have carried out.
